src/Components/Components/test/Routing/RouteTableFactoryTests.cs

```diff
@@ -366,6 +366,41 @@ public void ProducesAStableOrderForNonAmbiguousRoutes()
             Assert.Equal("a/brilliant", routeTable.Routes[0].Template.TemplateText);
         }
 
+        [Fact]
+        public void DoesNotThrowIfStableSortComparesRouteWithItself()
+        {
+            // Test for https://github.com/aspnet/AspNetCore/issues/13313
+            // Arrange & Act
+            var builder = new TestRouteTableBuilder();
```

Comment:
The issue specifically reproduces when you have [more than 16](https://github.com/dotnet/corefx/blob/master/src/Common/src/CoreLib/System/Collections/Generic/ArraySortHelper.cs#L28) elements to be sorted. I programatically generated this sequence that would cause an instance to be compared against itself.